Sources/TwitterTextEditor/TextViewDelegateForwarder.swift

```diff
@@ -0,0 +1,101 @@
+ If we have class that implement one of  it from overloaded methods, we can get the selector from the class.
+ */
+private class OverloadedSelectorURL: NSObject, UITextViewDelegate {
+    func textView(_ textView: UITextView, shouldInteractWith URL: URL, in characterRange: NSRange) -> Bool {
+        false
+    }
+
+    func textView(_ textView: UITextView, shouldInteractWith URL: URL, in characterRange: NSRange, interaction: UITextItemInteraction) -> Bool {
+        false
+    }
+}
+
+private class OverloadedSelectorTextAttachment: NSObject, UITextViewDelegate {
+    func textView(_ textView: UITextView, shouldInteractWith textAttachment: NSTextAttachment, in characterRange: NSRange) -> Bool {
+        false
+    }
+
+    func textView(_ textView: UITextView, shouldInteractWith textAttachment: NSTextAttachment, in characterRange: NSRange, interaction: UITextItemInteraction) -> Bool {
+        false
+    }
+}
+
+private let textViewDelegateSelectors: [Selector] = [
+    #selector(UITextViewDelegate.textView(_:shouldChangeTextIn:replacementText:)),
+    #selector(OverloadedSelectorURL.textView(_:shouldInteractWith:in:)),
+    #selector(OverloadedSelectorURL.textView(_:shouldInteractWith:in:interaction:)),
+    #selector(OverloadedSelectorTextAttachment.textView(_:shouldInteractWith:in:)),
+    #selector(OverloadedSelectorTextAttachment.textView(_:shouldInteractWith:in:interaction:)),
+    #selector(UITextViewDelegate.textViewDidBeginEditing(_:)),
+    #selector(UITextViewDelegate.textViewDidChange(_:)),
+    #selector(UITextViewDelegate.textViewDidChangeSelection(_:)),
+    #selector(UITextViewDelegate.textViewDidEndEditing(_:)),
+    #selector(UITextViewDelegate.textViewShouldBeginEditing(_:)),
+    #selector(UITextViewDelegate.textViewShouldEndEditing(_:)),
+    #selector(TextViewDelegate.textViewDidUnmarkText(_:)),
+    #selector(TextViewDelegate.textView(_:didChangeBaseWritingDirection:forRange:))
+]
+
+class TextViewDelegateForwarder: NSObject, TextViewDelegate {
+    weak var scrollViewDelegate: UIScrollViewDelegate?
+    weak var textViewDelegate: TextViewDelegate?
+
+
+    override func responds(to aSelector: Selector!) -> Bool {
+        if scrollViewDelegateSelectors.contains(aSelector) {
+            return scrollViewDelegate?.responds(to: aSelector) ?? false
+        }
+        if textViewDelegateSelectors.contains(aSelector) {
+            return textViewDelegate?.responds(to: aSelector) ?? false
+        }
+        return super.responds(to: aSelector)
+    }
+
+    override func forwardingTarget(for aSelector: Selector!) -> Any? {
+        if scrollViewDelegateSelectors.contains(aSelector) {
+            return scrollViewDelegate
+        }
+        if textViewDelegateSelectors.contains(aSelector) {
+            return textViewDelegate
+        }
+        return super.forwardingTarget(for: aSelector)
+    }
```

Comment:
I think we can change from `forwadingTarget(for:)` implementation to each delegate inline and forwarding in each delegate method, and it make better performance.
But I think we can't change `responds(to:)`, it decide delegate method call or not, if we remove it, every delegate method should be called. (even if user don't implement delegate method..)
And not so affect to performance, because many of delegate method is checked `responds(to:)` when delegate object assigned. (not n time called)

Or, making class dynamically? (I feel it's too much hacking 😅)

So removing `forwadingTarget(for:)` and implement each delegate method as forwarding, and keep `responds(to:)` implementation is looks nice. How about it?

I will measure the performance if changing `forwardingTarget(for:)` later.

Reply:
That’s true, all of them are `@optional` and `responds(to:)` costs more.
It is interesting me to try removing `forwardingTarget(for:)` but probably, overall, wouldn't be affect too.
Wondering if you’ve tried and measure the performance.

Either way, I am going to test current patch and see how it goes in the application to ensure the patch works.

Reply:
Sorry for forgot them.
This is my experiment code https://gist.github.com/tarunon/f30c9023c6386248c9749f6f06ada461.
The result is a difference of about 1.5 times, looks quite large, but the overhead of the function call is very small (50ms difference in 100,000 times), so I don't know if this change will be worth it.